### PR TITLE
feat(dashboard): Add a JSONField to store widget details

### PR DESCRIPTION
**Pull request by shruthilayaj**
We have a few columns on our DashboardWidget model
that are *descriptors* of sorts for what a dashboard
widget might look like, for example: `display_type`,
`widget_type`, `interval`...

We anticipate adding more descriptors in the future,
like widget sizes, maybe widget coordinates, table
widget column widths, top events limit etc. and it
would be nice to not have to create a field and
migration for each of these on this model.
Instead, we can put these in a JSON field.

### CODE REVIEW

File: src/sentry/migrations/0265_add_dashboard_widget_detail_field.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not.
+    # By default we prefer to run in a transaction, but for migrations where you want
+    # to `CREATE INDEX CONCURRENTLY` this needs to be set to False. Typically you'll
+    # want to create an index concurrently when adding one to an existing table.
+    # You'll also usually want to set this to `False` if you're writing a data
+    # migration, since we don't want the entire migration to run in one long-running
+    # transaction.
+    atomic = True
```
Comment (shruthilayaj) [inline src/sentry/migrations/0265_add_dashboard_widget_detail_field.py:27]: This table isn't too big (not sure if I should post the # of rows here)

**Comment by markstory** [comment]
> We anticipate adding more descriptors in the future, like widget sizes, maybe widget coordinates, table widget column widths, top events limit etc. and it would be nice to not have to create a field and migration for each of these on this model. Instead, we can put these in a JSON field.

Do you see the need to filter or sort on any of the fields that are moving or being added to the JSON blob?

**Review by markstory — APPROVED** [review]
Schema change looks good to me.

**Review by wedamija — COMMENTED** [review]
I'm generally against adding a json blob when it's just for the sake of convenience. Using columns allows you to enforce data types better and is more easily queryable. Not blocking the change, but it's worth considering whether this is the right choice moving forward.

**Comment by shruthilayaj** [comment]
@wedamija @markstory We're going to be adding widget metadata that may not apply to all widgets for example table column width or top n limits. We're never really going to filter/query on any of these fields - they are just used to render the widget data

**Comment by wedamija** [comment]
> @wedamija @markstory We're going to be adding widget metadata that may not apply to all widgets for example table column width or top n limits. We're never really going to filter/query on any of these fields - they are just used to render the widget data. Kinda like the discover saved query model

For dynamic fields this makes sense, is it worth considering making columns for anything that is common across all widget types?

**Comment by shruthilayaj** [comment]
> > @wedamija @markstory We're going to be adding widget metadata that may not apply to all widgets for example table column width or top n limits. We're never really going to filter/query on any of these fields - they are just used to render the widget data. Kinda like the discover saved query model
> 
> For dynamic fields this makes sense, is it worth considering making columns for anything that is common across all widget types?

Okay cool @wedamija, so I'm going to add the `detail` field and close the follow up backfill PR, because the only dynamic field I'd want to backfill is `interval` which is the rollup for chart data (from what I can tell), but it feels like that should just be dynamically calculated based on the time range in the global selection header and the chart type- not sure why it needs to be stored in the database (will revisit that). And we'll put the dynamic fields we want to introduce into this JSON blob and common fields as a model field.

**Review by wedamija — APPROVED** [review]

**Pull request merged**